modules/issue_tracker/php/endpoints/attachment.class.inc

```diff
@@ -0,0 +1,259 @@
+<?php declare(strict_types=1);
+/**
+ * POST request methods.
+ *
+ * Used to handle issue_tracker attachment requests.
+ *
+ * PHP Version 7
+ *
+ * @category LORIS_Module
+ * @package  Login
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\issue_tracker\endpoints;
+
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use \LORIS\Middleware\ETagCalculator;
+use \LORIS\api\Endpoint;
+
+/**
+ * The NDB_Page for the POST requests.
+ *
+ * @category Loris
+ * @package  IssueTracker
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class Attachment extends Endpoint implements ETagCalculator
+{
+    public $skipTemplate = true;
+
+    /**
+     * This function will return a json object
+     * for the specific attachment request.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Ensure GET or POST request.
+        switch ($request->getMethod()) {
+        case 'GET':
+            return $this->_handleGET($request);
+        case 'POST':
+            return $this->_handlePOST($request);
+        case 'DELETE':
+            return $this->_handleDELETE($request);
+        default:
+            return new \LORIS\Http\Response\JSON\MethodNotAllowed(
+                $this->allowedMethods()
+            );
+        }
+    }
+
+    /**
+     * Handle GET requests for attachment
+     * and used for downloading attachment.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleGET(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Parse GET query params.
+        $values = $request->getQueryParams();
+
+        $file_hash = $values['file_hash'];
+        $filename  = $values['filename'];
+        $mime_type = $values['mime_type'];
+
+        $config = \NDB_Config::singleton();
+        $attachment_data_dir = rtrim(
+            $config->getSetting('IssueTrackerDataPath'),
+            '/'
+        );
+
+        $fileToDownload = trim($attachment_data_dir) .
+            '/attachments/' .
+            $file_hash;
+
+        if (!is_readable($fileToDownload)) {
+            throw new \LorisException('Attachment NOT FOUND');
```

Comment:
This message doesn't match the error condition. I think this should be a Forbidden message, not an Exception.

Reply:
@johnsaigle updated the message.